gef.py

```diff
@@ -8658,81 +8658,97 @@ def get_settings_path(self):
         return path if os.path.isdir(path) else None
 
 
-class GenericOffsetFunction(gdb.Function):
-    """This is an abstract class for invoking offset functions, should not be instantiated."""
+@lru_cache()
+def get_section_base_address(name):
+    section = process_lookup_path(name)
+    if section:
+        return section.page_start
+
+    return None
+
+@lru_cache()
+def get_zone_base_address(name):
+    zone = file_lookup_name_path(name, get_filepath())
+    if zone:
+        return zone.zone_start
+
+    return None
+
+class GenericFunction(gdb.Function):
+    """This is an abstract class for invoking convenience functions, should not be instantiated."""
     __metaclass__ = abc.ABCMeta
 
-    @abc.abstractproperty
-    def _section_(self): pass
-    @abc.abstractproperty
-    def _zone_(self): pass
     @abc.abstractproperty
     def _function_(self): pass
     @property
     def _syntax_(self):
         return "${}([offset])".format(self._function_)
 
     def __init__ (self):
-        super(GenericOffsetFunction, self).__init__(self._function_)
+        super(GenericFunction, self).__init__(self._function_)
 
-    def invoke(self, offset=gdb.Value(0)):
+    def invoke(self, *args):
         if not is_alive():
```

Comment:
Why not use `@only_if_gdb_running`?

Reply:
I'm guessing something to do with this being a function and not a command means you have to raise a `GdbError`. If that's the case, cool.

Reply:
`@only_if_gdb_running` prints a warning and will return `None` instead of throwing a gdb.Error so you get something like:
```
[*] No debugging session active
Python Exception <class 'TypeError'> Could not convert Python object: None.: 
Error while executing Python code.
```

But it could go on the `do_invoke` method and  `invoke` could check for `None`

Reply:
Naw this is fine.